packages/flutter_tools/lib/src/commands/analyze_base.dart

```diff
@@ -235,7 +206,6 @@ class PackageDependencyTracker {
           throwToolExit('pubspec.yaml is malformed.');
         }
       }
-      dependencies.addDependenciesFromPackagesFileIn(directory);
     }
 
     // prepare a union of all the .packages files
```

Comment:
nit. update this comment, s/.packages/pubspec.yaml?

Reply:
I don't see how the code is preparing a union of anything here. I think the comment is just broken.

This comment seems to have appeared in https://github.com/flutter/flutter/pull/6093
It might have been mistakenly copied from [here](https://github.com/flutter/flutter/commit/a0227cab15e230a564fb6316347b8e0ce3d21f6c#diff-a6e4bf25b45ecb000db25d7343a44f018ff5302d4695eaa1277d804d2252c1f5R148) (where it makes a lot more sense).

To minimize scope of the PR I'll leave the comment as-is here, and make a follow-up removing it.